### PR TITLE
Play nicely with private DockerHub registries

### PR DESCRIPTION
**Pull request by shamil**
**The problem**

`Fn` looks for domain name in the image itself. And then checks if there is a **suffix** with that domain in the auth config file: https://github.com/fnproject/fn/blob/5dc5740a541cb5a3aba4316ef0295e9eefe3b936/api/agent/drivers/docker/docker.go#L318

So to make it work I ended up to name my images like so `docker.io/<myorg>/<myimage>:<tag>`

Then in my auth I made a change of my registry URL from `https://index.docker.io/v2/` to the domain which ends with `docker.io`, like so

```{"auths":{"https://index.docker.io":{"auth":"yourauthbase64here"}}}```

***

**So to sum it up**

1. `Fn` doesn't know how to work with dockerhub private registries out of the box
2. To validate domain against image name, it uses `Suffix` check, which not always ends with domain name.

This PR should fix both problems.

### CODE REVIEW

**Comment by shamil** [comment]
closed in favor of #1130
